src/main/kotlin/controllers/worldRenderer/Renderer.kt

```diff
@@ -46,14 +48,17 @@ import org.slf4j.LoggerFactory
 import java.awt.event.ActionListener
 import java.nio.FloatBuffer
 import java.nio.IntBuffer
+import javax.inject.Singleton
 import kotlin.math.min
 
+@Singleton
```

Comment:
Should instead create a provider class for this in the injector configuration (see ModelsModule, CacheModule and LoaderModule for examples)

The renderer should only be a singleton because the injector says so, not by some more artificial constraint here.

Reply:
Can you show me what this looks like?